.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,100 @@
+# Modified from: https://github.com/oats-center/isoblue/blob/master/.github/workflows/build-and-push-services.yml
+name: Build and Push Pong Containers
+
+on:
+  push:
+  pull_request:
+
+env:
+  REGISTRY: ghcr.io
+  ORG: aaron97neu
+  CONTAINER_DIR: src
+
+jobs:
+  build-and-push-services:
+    strategy:
+      matrix:
+        service:
+          - ai-paddle-control
+          - audio-engine
+          - clocktower-visualizer
+          - game-engine
+          - gameboard
+          - human-paddle-control
+          - human-visualizer
+          - neural-net-visualizer
+          
+    runs-on: ubuntu-latest
+   
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in this job.
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3.7.1
+        with:
+          version: latest
+
+      - name: Builder instance name
+        run: echo ${{ steps.buildx.outputs.name }}
+
+      - name: Available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3.3.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          flavor: |
+            latest=false
+          bake-target:
+            ${{ matrix.service }}
+          images: |
+            ${{ env.REGISTRY }}/${{ env.ORG }}/${{ matrix.service }}
+          tags: |
+            type=sha
+            type=semver,pattern={{version}},enable=${{ startsWith(github.ref, format('refs/heads/{0}', 'release')) }}
+            type=semver,pattern={{major}}.{{minor}},enable=${{ startsWith(github.ref, format('refs/heads/{0}', 'release')) }}
+            type=semver,pattern={{major}},enable=${{ startsWith(github.ref, format('refs/heads/{0}', 'release')) }}
+            type=ref,event=branch
+            type=ref,event=pr
+            type=raw,value=next,enable={{is_default_branch}}
+          labels: |
+            org.opencontainers.image.title=${{ matrix.service }}
+
+      - name: Print out bake files for debugging
+        run: |
+          echo Bake File:
+          cat ${{ steps.meta.outputs.bake-file }}
+          echo -e "\nBake File Annotations:"
+          cat ${{ steps.meta.outputs.bake-file-annotations }}
+          echo -e "\nBake File tags:"
+          cat ${{ steps.meta.outputs.bake-file-tags }}
```

Comment:
do you have an example of what this output looks like?

Reply:
Yep, expand the "Print out bake files for debugging" section here:
https://github.com/aaron97neu/Discovery-World-Pong/actions/runs/11960723274/job/33345366795

<img width="906" alt="image" src="https://github.com/user-attachments/assets/8a0d1fbd-ff34-4506-980b-0492f83ec90e">